README.md

```diff
@@ -76,27 +76,41 @@ This meta-role requires a permission policy similar to the following:
-| role_description | The description to associate with the IAM role (as well as the corresponding policy) that allows read-only access to the specified object(s) in the specified S3 buckets.  Note that the first "%s" in this value will get replaced with the s3_bucket variable and the second "%s" will get replaced with the entity_name variable. | `string` | `Allows read-only access to S3 bucket %s required for %s.` | no |
-| role_name | The name to assign the IAM role (as well as the corresponding policy) that allows read-only access to the specified S3 buckets.  Note that the first "%s" in this value will get replaced with the s3_bucket variable and the second "%s" will get replaced with the entity_name variable. | `string` | `%s-ReadOnly-%s` | no |
-| role_tags | Tags to apply to the IAM role that allows read-only access to the specified S3 buckets. | `map(string)` | `{}` | no |
-| s3_bucket | The name of the S3 bucket that the created role will be allowed to read from (e.g. "my-bucket"). | `string` | n/a | yes |
-| s3_objects | A list specifying the objects in the S3 bucket that the created role will be allowed to read (e.g. ["my-file", "projects\example\*"]).  AWS-supported S3 ARN wildcards (* and ?) can be used, but full regular expressions can not.  If not specified, the role will be allowed to read any object in the bucket. | `list(string)` | `["*"]` | no |
+| account\_ids | AWS account IDs that are allowed to assume the role. | `list(string)` | `[]` | no |
+| aws\_region | The AWS region where the non-global resources are to be provisioned (e.g. "us-east-1"). | `string` | `"us-east-1"` | no |
+| entity\_name | The name of the entity that the role is being created for (e.g. "test-user"). | `string` | n/a | yes |
+| iam\_usernames | The list of IAM usernames allowed to assume the role.  If not provided, defaults to allowing any user in the specified account(s).  Note that including "root" in this list will override any other usernames in the list. | `list(string)` | `["root"]` | no |
+| role\_description | The description to associate with the IAM role (as well as the corresponding policy) that allows read-only access to the specified object(s) in the specified S3 buckets.  Note that the first "%s" in this value will get replaced with the s3\_bucket variable and the second "%s" will get replaced with the entity\_name variable. | `string` | `"Allows read-only access to S3 bucket %s required for %s."` | no |
+| role\_name | The name to assign the IAM role (as well as the corresponding policy) that allows read-only access to the specified S3 buckets.  Note that the first "%s" in this value will get replaced with the s3\_bucket variable and the second "%s" will get replaced with the entity\_name variable. | `string` | `"%s-ReadOnly-%s"` | no |
+| role\_tags | Tags to apply to the IAM role that allows read-only access to the specified S3 buckets. | `map(string)` | `{}` | no |
```

Comment:
Is this being used in the same way that the `tags` variable is used in other TF repositories? This usage seems contrary to the changes you started in the skeleton:
https://github.com/cisagov/s3-read-role-tf-module/blob/8339b8eef83c40d444b9062dd751f5efc055cc3e/role.tf#L24-L30

Reply:
It's a slightly different use, since the `role_tags` are only applied to a specific resource.  I left this behavior in place because I thought changing it would interfere with the way certain resources are labeled for use by the test users in [cisagov/skeleton-ansible-role-with-test-user](https://github.com/cisagov/skeleton-ansible-role-with-test-user) or [cisagov/skeleton-packer](https://github.com/cisagov/skeleton-packer), for example.  Perhaps @dav3r could comment on this aspect.

Even if that is the case, it might make more sense to rename this input `additional_tags_for_role` or something similar, since whatever tags are included in the variable will now be applied _in addition to_ the default tags with which the provider is configured.  I think it makes sense to make this change in a separate PR, though, since it will break a lot of other repos.

TL;DR - I was afraid that changing `role_tags` would break the AWS test users so I left it in place.  I agree that the variable needs to be renamed, at a minimum.

Reply:
I was skimming over [repos that use this module](https://github.com/search?q=org%3Acisagov+s3-read-role-tf-module&type=code) and it looks like it's being called in a way that the individual Ansible roles would resolve just by switching to using `default_tags` in their providers. In [cisagov/terraform-state-read-role-tf-module](https://github.com/cisagov/terraform-state-read-role-tf-module) it looks like `role_vars` is mainly being passed through to remove `Workspace` tags in calling Terraform projects.

Reply:
I like @jsf9k's idea of renaming `role_tags` to `additional_tags_for_role` to make things a bit clearer.  That will allow those tags to be applied to the role created by this repo, while the default tags (from the provider) will be applied to both the role and the policy (and any other taggable resources that are added in the future).

Reply:
I realized in debugging `terraform apply` errors in cisagov/ansible-role-cobalt-strike#35 that Terraform treats the case where `tags` is equal to the provider's `default_tags` as an error.  This means that this module is in some cases borked as it stands.  See [here](https://github.com/cisagov/ansible-role-cobalt-strike/blob/f93638345a3dea3fc531071ecb78796f74147e75/terraform/bucket_roles.tf#L21-L33), for instance.

I will go ahead and merge this PR, then create a new one to make the change I described [here](https://github.com/cisagov/s3-read-role-tf-module/pull/14#discussion_r651194471), which will resolve the borking.

Reply:
> I'd like to go ahead and merge this PR, then create a new one to make the change I described [here](https://github.com/cisagov/s3-read-role-tf-module/pull/14#discussion_r651194471), which will resolve the borking. Are you OK with that @dav3r and @mcdonnnj?

I'm good with that plan.

Reply:
> I'd like to go ahead and merge this PR, then create a new one to make the change I described here, which will resolve the borking. Are you OK with that @dav3r and @mcdonnnj?

Won't changing the name from `role_vars` to `additional_tags_for_role` result in the same borking? If the values are the same as in `default_tags` it won't matter what we name it right?

Reply:
@mcdonnnj - yes, but I will only pass the _additional_ (non-`default-tags`) tags to `additional_tags_for_role`.  I will have to make PRs for all the repos that use this module.

I will also add a `tags_to_ignore_for_role` parameter and use the `lifecycle --> ignore_changes` foo to ignore changes to those particular tags.  That won't stop the `Workspace` tag, for example, from being applied to the role, but it will stop it from changing.  If you or @dav3r can think of a better way to handle this case then let me know.  An alternative is to add the `Workspace` manually everywhere else via `tags`, but I don't like that solution either; maybe you guys do though.

Reply:
Ah, ok. The second half is the part I was missing. That makes sense as a combined change. As far as `Workspace` tag should the caller be the one to worry about it as done [here](https://github.com/cisagov/cool-images-assessment-images/blob/8c2474e6ecbd0a097778f809ec98e2094e03b7e5/read_terraform_state_role.tf#L23)?

Reply:
> Ah, ok. The second half is the part I was missing. That makes sense as a combined change. As far as `Workspace` tag should the caller be the one to worry about it as done [here](https://github.com/cisagov/cool-images-assessment-images/blob/8c2474e6ecbd0a097778f809ec98e2094e03b7e5/read_terraform_state_role.tf#L23)?

If the tag they want to ignore is one of the additional tags then they could remove it themselves; but, if it's one of the provider's default tags then they wouldn't be able to remove it like that.

Reply:
@jsf9k - I cannot think of a better solution here than what you have come up with.  I say go for it.